frontend/apps/filemanager/filemanagermenu.lua

```diff
@@ -213,6 +213,13 @@ function FileManagerMenu:setUpdateItemTable()
                                 purgeDir(cachedir)
                             end
                             lfs.mkdir(cachedir)
+                            -- Clear Cache object, so it doesn't serialize
+                            -- anything old on exit
+                            local Cache = require("cache")
+                            Cache:clear()
```

Comment:
I think this is not going to be safe because `clear` calls `onFree` method on all the cached objects. Some of those objects are allocated through malloc, which means those references will have a dangling pointer to a freed memory. It's really hard to do this right without restarting the reader from scratch. I still think we should create a prompt for confirmation before the purge and force user to restart the reader after the purge.